trucks.js

```diff
@@ -501,4 +501,77 @@ var foodTrucks = [
     Facebook: 'https://www.facebook.com/fulltilticecreams',
     Twitter: 'https://twitter.com/fticecream'
   }
-];
\ No newline at end of file
+];
+
+function getTrucks(){
+  return foodTrucks;
+}
+
+function getTruck(name){
+  return foodTrucks.find(function(truck){
+    return truck.name === name;
+  });
+}
+
+function getFoodTypes(){
+  var types = foodTrucks.map(function (truck){
+    return truck.type;
+  });
+  types = _.flatten(types);
+  types = _.uniq(types);
+  
+  return types;
+}
+
+function filterTrucksByDay(day){
+  return foodTrucks.filter(function (truck){
+    return truck.schedule.indexOf(day) > -1;
+  });
+}
+
+function filterTrucksByFoodType(foodType){
+  return foodTrucks.filter(function (truck){
+    var types = truck.type.map(function (type){
+      return type.toLowerCase();
+    });
+    
+    return types.indexOf(foodType.toLowerCase()) > -1;
+  });
+}
+
+function Truck(name,
+  type,
+  schedule,
+  payment,
+  description,
+  Facebook){
+   Truck.name= name,
+   Truck.type= type,
+   Truck.schedule= schedule,
+   Truck.payment= payment,
+   Truck.description= description,
+   Truck.Facebook= Facebook
+};
+
+var truck = new Truck();
+
+function addTruck(truck){
+  foodTrucks.push(truck);
+  
+  return truck;
+};
+
+function removeTruck(name){
+  var theTruck = foodTrucks.findIndexOf(name);
+  delete foodTrucks[theTruck];
```

Comment:
deleting something from an array like this will leave the spot in the array and assign to it `undefined` which is not what you want. Instead, look at the documentation for how to use splice to remove something from an array.